src/driver/postgres/PostgresQueryRunner.ts

```diff
@@ -711,21 +711,24 @@ export class PostgresQueryRunner extends BaseQueryRunner implements QueryRunner
             if (
                 (newColumn.type === "enum" || newColumn.type === "simple-enum")
                 && (oldColumn.type === "enum" || oldColumn.type === "simple-enum")
-                && !OrmUtils.isArraysEqual(newColumn.enum!, oldColumn.enum!)
+                && (!OrmUtils.isArraysEqual(newColumn.enum!, oldColumn.enum!)
+                    || oldColumn.enumName !== newColumn.enumName)
             ) {
-                const enumName = this.buildEnumName(table, newColumn);
                 const arraySuffix = newColumn.isArray ? "[]" : "";
-                const oldEnumName = this.buildEnumName(table, newColumn, true, false, true);
-                const oldEnumNameWithoutSchema = this.buildEnumName(table, newColumn, false, false, true);
-                const enumTypeBeforeColumnChange = await this.getEnumTypeName(table, oldColumn);
+                const {enumTypeSchema: prevEnumSchema, enumTypeName: prevEnumName } = await this.getEnumTypeName(table, oldColumn);
+                const prevEnumNameWithOldSuffix = this.buildEnumName(table, oldColumn, false, false, true);
+                const prevEnumNameWithOldSuffixAndSchema = this.buildEnumName(table, oldColumn, true, false, true);
+                const newEnumName = this.buildEnumName(table, newColumn);
+                const newEnumNameWithOldSuffix = this.buildEnumName(table, newColumn, false, false);
+                const newEnumNameWithOldSuffixAndSchema = this.buildEnumName(table, newColumn, true, false, true);
 
                 // rename old ENUM
-                upQueries.push(new Query(`ALTER TYPE "${enumTypeBeforeColumnChange.enumTypeSchema}"."${enumTypeBeforeColumnChange.enumTypeName}" RENAME TO ${oldEnumNameWithoutSchema}`));
-                downQueries.push(new Query(`ALTER TYPE ${oldEnumName} RENAME TO  "${enumTypeBeforeColumnChange.enumTypeName}"`));
+                upQueries.push(new Query(`ALTER TYPE "${prevEnumSchema}"."${prevEnumName}" RENAME TO ${prevEnumNameWithOldSuffix}`));
+                downQueries.push(new Query(`ALTER TYPE ${newEnumName} RENAME TO ${newEnumNameWithOldSuffix}`));
 
                 // create new ENUM
                 upQueries.push(this.createEnumTypeSql(table, newColumn));
-                downQueries.push(this.dropEnumTypeSql(table, oldColumn));
+                downQueries.push(this.createEnumTypeSql(table, oldColumn));
```

Comment:
looks like a typo?

Reply:
whoops thanks for pointing it out!

Reply:
so actually I think that this isn't wrong - when you're doing a `down` query we want to create the old enum type, not delete it - because the old enum type doesn't exist in the database yet. What do you think?

Reply:
down queries runs in reverse order. In this case `dropEnumTypeSql` will be executed at the end of Down process and drops a "new" enum type, which was created by Up query. So if you replace it with `createEnumTypeSql` it will try to create already existiing "new" enum type

Reply:
Oh I see didn’t know that, will fix then!